src/lib/UserSetting/AutosaveEnabled.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\UserSetting;
+
+use EzSystems\EzPlatformUser\UserSetting\FormMapperInterface;
+use EzSystems\EzPlatformUser\UserSetting\ValueDefinitionInterface;
+use JMS\TranslationBundle\Annotation\Desc;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+use eZ\Publish\Core\Base\Exceptions\InvalidArgumentException;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+
+class AutosaveEnabled implements ValueDefinitionInterface, FormMapperInterface
+{
+    public const ENABLED_OPTION = 'enabled';
+    public const DISABLED_OPTION = 'disabled';
+
+    /** @var \Symfony\Contracts\Translation\TranslatorInterface */
+    private $translator;
+
+    public function __construct(
+        TranslatorInterface $translator
+    ) {
+        $this->translator = $translator;
+    }
+
+    public function getName(): string
+    {
+        return $this->getTranslatedName();
+    }
+
+    public function getDescription(): string
+    {
+        return $this->getTranslatedDescription();
+    }
+
+    public function getDisplayValue(string $storageValue): string
+    {
+        switch ($storageValue) {
+            case self::ENABLED_OPTION:
```

Comment:
The class/user setting name is confusing and does not go well with choice options. I don't think it's optimal to actually store "enabled/disabled" states. Simple `bool` value (more likely 0/1 since we store strings) would be sufficient and more appropriate. It would represent simple on/off checkbox more precisely.

`autosave_enabled: enabled/disabled` is counter intuitive, should be `true/false`

Reply:
I have renamed option to `autosave`.